components/raftstore/src/store/peer_storage.rs

```diff
@@ -1085,26 +1085,33 @@ impl PeerStorage {
     }
 
     /// Check if the storage is applying a snapshot.
+    ///
+    /// When it returns:
+    /// `None` means no snapshot is being applied at all or the snapshot is cancelled,
+    /// `Some(true)` means a snapshot is being applied,
+    /// `Some(false)` means a snapshot is just applied.
     #[inline]
-    pub fn check_applying_snap(&mut self) -> bool {
+    pub fn check_applying_snap(&mut self) -> Option<bool> {
```

Comment:
Option<bool> as a return value is a confusing design, use a enum would be more readable.